scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -0,0 +1,335 @@
+
+    // Get the number of SMRT Cells in the run directory
+    dto.setLaneCount((int) getCellDirectory(runDirectory).count());
+
+    // Presence of Transfer_Test_*.txt indicates the run has started, get the creation time
+    getMetadataDirectory(runDirectory)
+        .flatMap(
+            metadataDirectory ->
+                Arrays.stream(
+                    metadataDirectory.listFiles(file -> TRANSFER_TEST.test(file.getName()))))
+        .forEach(
+            s -> {
+              try {
+                FileTime fileTime =
+                    (FileTime) Files.getAttribute(Path.of(s.getAbsolutePath()), "creationTime");
+                // Get the earliest time to use as run start date
+                if (dto.getStartDate() == null
+                    || fileTime.toInstant().isBefore(dto.getStartDate())) {
+                  dto.setStartDate(fileTime.toInstant());
+                }
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            });
+
+    // We are good to process this run
+    getMetadataDirectory(runDirectory)
+        .flatMap(
+            metadataDirectory ->
+                Arrays.stream(
+                    metadataDirectory.listFiles(file -> file.getName().endsWith(".metadata.xml"))))
+        .map(RunProcessor::parseXml)
+        .filter(Optional::isPresent)
+        .forEach(metadata -> processMetadata(metadata.get(), dto, tz));
+
+    // Check for .transferdone and Transfer_Test in all SMRT Cells to consider the run complete
+    if (getMetadataDirectory(runDirectory)
+                .flatMap(
+                    metadataDirectory ->
+                        Arrays.stream(
+                            metadataDirectory.listFiles(
+                                file -> TRANSFER_DONE.test(file.getName()))))
+                .count()
+            == getCellDirectory(runDirectory).count()
+        && getMetadataDirectory(runDirectory)
+                .flatMap(
+                    metadataDirectory ->
+                        Arrays.stream(
+                            metadataDirectory.listFiles(
+                                file -> TRANSFER_TEST.test(file.getName()))))
+                .count()
+            == getCellDirectory(runDirectory).count()) {
+      // We have all the expected files, the run is considered finished, get the completion time
+      dto.setHealthType(HealthType.COMPLETED);
+      getMetadataDirectory(runDirectory)
+          .flatMap(
+              metadataDirectory ->
+                  Arrays.stream(
+                      metadataDirectory.listFiles(file -> TRANSFER_DONE.test(file.getName()))))
+          .forEach(
+              s -> {
+                try {
+                  FileTime fileTime =
+                      (FileTime) Files.getAttribute(Path.of(s.getAbsolutePath()), "creationTime");
+                  // Get the latest file creation time for completion date
```

Comment:
Could similarly use a `.max` function on the stream, and then set the DTO's completionDate